Packages/com.unity.inputsystem/InputSystem/Actions/InputActionMap.cs

```diff
@@ -836,6 +836,23 @@ private unsafe void SetUpPerActionCachedBindingData()
                 m_SingletonAction.m_BindingsCount = m_Bindings.Length;
                 m_SingletonAction.m_ControlStartIndex = 0;
                 m_SingletonAction.m_ControlCount = m_State?.totalControlCount ?? 0;
+
+                // Only complication, InputActionState allows a control to appear multiple times
+                // on the same action and InputAction.controls[] doesn't.
+                if (m_ControlsForEachAction.HaveDuplicateReferences(0, m_SingletonAction.m_ControlCount))
+                {
+                    var numControls = 0;
+                    var controls = new InputControl[m_SingletonAction.m_ControlCount];
+                    for (var i = 0; i < m_SingletonAction.m_ControlCount; ++i)
```

Comment:
On this for loop I would suggest using braces to increase readability and reduce likelyhood of future error

Reply:
Done